src/overlays/actors/ovl_En_Fg/z_en_fg.c

```diff
@@ -19,42 +41,417 @@ const ActorInit En_Fg_InitVars = {
+        { 0x000010AA, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON | BUMP_HOOKABLE,
+        OCELEM_ON,
+    },
+    { 8, 10, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    1, 0, 0, 0, MASS_IMMOVABLE,
+};
+
+static DamageTable sDamageTable = {
+    0x00, 0x11, 0x00, 0x01, 0x00, 0xE1, 0x11, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x31, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D280.asm")
+static EnFgAnimation sAnimations[] = {
+    { &D_06001534, 1.0f, 0, -1, 0, 0 },
+    { &D_06001534, 1.0f, 0, -1, 0, -4 },
+    { &D_060011C0, 1.0f, 0, -1, 0, -4 },
+    { &D_060007BC, 1.0f, 0, -1, 2, -4 },
+};
+
+s32 func_80A2D280(SkelAnime* skelAnime, s16 animIndex) {
+    s16 frameCount;
+    s32 ret;
+
+    ret = false;
+    if (animIndex >= 0 && animIndex < 4) {
+        ret = true;
+        frameCount = sAnimations[animIndex].frameCount;
+        if (frameCount < 0) {
+            frameCount = SkelAnime_GetFrameCount((GenericAnimationHeader*)sAnimations[animIndex].animationSeg);
+        }
+        SkelAnime_ChangeAnim(skelAnime, sAnimations[animIndex].animationSeg, sAnimations[animIndex].playbackSpeed,
+                                sAnimations[animIndex].frame, frameCount, sAnimations[animIndex].mode,
+                                sAnimations[animIndex].transitionRate);
+    }
+    return ret;
+}
+
+void func_80A2D348(EnFg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health != 0) {
+        this->collider.dim.pos.x = this->actor.world.pos.x;
+        this->collider.dim.pos.y = this->actor.world.pos.y;
+        this->collider.dim.pos.z = this->actor.world.pos.z;
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+    }
+}
+
+void func_80A2D3D4(EnFg* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+u8 func_80A2D400(EnFg* this) {
+    if (this->actor.colChkInfo.damage >= this->actor.colChkInfo.health) {
+        this->actor.colChkInfo.health = 0;
+    } else {
+        this->actor.colChkInfo.health = this->actor.colChkInfo.health - this->actor.colChkInfo.damage;
+    }
+    return this->actor.colChkInfo.health;
+}
+
+s32 func_80A2D42C(EnFg* this) {
+    s32 ret = 0;
+
+    if (this->collider.base.acFlags & 2) {
+        switch (this->actor.colChkInfo.damageEffect) {
+            case 1:
+                ret = 2;
+                break;
+            case 15:
+                ret = 3;
+                break;
+            case 14:
+                ret = 4;
+                break;
+            case 3:
+                ret = 5;
+                break;
+            default:
+                ret = 1;
+                break;
```

Comment:
Could these return values be an enum for each of the different frogs at the different locations / the different colors?
Like Laundry_Pool, Southern_Swamp, Woodfall_Temple, Mountain_Village, and Great_Bay_Temple?

Reply:
This file is strange because I don't think it's the primary frogs file, that file is En_Minifrog. This might be beta frogs, or cutscene frogs, I have no idea. 

I was able to figure out the order of the frogs based on the color data: `Yellow, Cyan, Pink, Blue, White`, but I wasn't able to work out if it should be enum from 0-4 or 1-5. The only time this enum is used is this switch function which is based on damage effects, which is somehow related to frogs? Poor froggies. (the enum is also used when this function is called in other switch functions).

If I had to guess, I'd say the enum would be 0-4 since there is a sixth black color available in the envColor data for the frogs, but I left out the enum since I wasn't fully confident in it. Thoughts?

Reply:
Document what you know in comments.
Since you weren't thorough in docs, I won't require the enum.

Reply:
I put the enum for the frogs in, even though they go basically unused. The switch here really doesn't seem to be related to which frog it is, it could apply to any of them